### PR TITLE
Further WattTime Integration, with Forecasting, and first Integration Tests

### PR DESCRIPTION
**Pull request by dnastrain**
Issue Number: #63 

## Summary
This PR adds the additional work done by Microsoft and UBS to integrate the Watttime features into the Carbon Aware SDK and to adds Integration Tests for the WebAPI

## Changes

- Please see the Gameplan and Azure Dev Ops for all comprehensive changes and development work.
- Update WattTime functionality
- Adds first Integration Tests for the Web API
- Add new API routes (Forcasting)
- Add local unit testing, local integration testing that also run during CICD

## Checklist

- [x] Local Tests Passing?
- [x] CICD and Pipeline Tests Passing?
- [x] Added any new Tests?
- [x] Documentation Updates Made?
- [ ] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?
Yes. This change refactors and reworks multiple aspects of the Carbon Aware SDK and will likely leave the repository in a state incompatible with prior versions. There is no easy workaround to this. 

## Anything else?
All development occurs in this fork: https://github.com/microsoft/carbon-aware-sdk

### CODE REVIEW

**Comment by dnastrain** [comment]
@vaughanknight , this is a work in progress, and we'll follow-up with you next week.

**Comment by dnastrain** [comment]
Hi @vaughanknight, this PR we consider ready for merge into the 0.1-preview branch. Happy to talk offline, if needed.

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<!-- BEGIN MICROSOFT SECURITY.MD V0.0.7 BLOCK -->
```
Comment (vaughanknight) [inline SECURITY.md:1]: We shouldn't have this file in the GSF public repo as it relates to the official Microsoft repo.

**Review by vaughanknight — CHANGES_REQUESTED** [review]
No major changes requested. 
* Please remove the SECURITY.MD file as it's part of the Microsoft repository policy but not for the GSF.
* Should ms-internal/azure-pipelines.yml be part of the project / PR?  Is yes, it will need a naming convention change from ms-internal?  Or is this specific to internal Microsoft needs - in this case it should be removed along with the folder?

Overall the changes look great.  In addition the quality of life changes such as the dotnet folder and tests moving into src folders are greatly appreciated.  I know there are some further changes coming to this PR which relate to bugs picked up in integration testing and will await those.  Look forward to those updates so we can move towards a 1.0 release once merged asap!

**Review by vaughanknight — APPROVED** [review]
Files all reviewed with changes updated as discussed.  Merging into 0.1-preview and looking to merge into dev branch soon.

**Pull request merged**